diff -ruN -x .git ../linux-ng/repo/drivers/input/touchscreen/atmel_mxt_ts.c ./repo/drivers/input/touchscreen/atmel_mxt_ts.c
--- ../linux-ng/repo/drivers/input/touchscreen/atmel_mxt_ts.c	2016-03-16 15:43:17.000000000 +0000
+++ ./repo/drivers/input/touchscreen/atmel_mxt_ts.c	2016-07-26 18:37:54.888808590 +0000
@@ -2487,6 +2487,14 @@
 	{ }
 };
 
+static struct mxt_acpi_platform_data peppy_platform_data[] = {
+	{
+		/* Touchscreen */
+		.hid	= "ATML0001",
+	},
+	{ }
+};
+
 static unsigned int chromebook_tp_buttons[] = {
 	KEY_RESERVED,
 	KEY_RESERVED,
@@ -2517,12 +2525,19 @@
 		/* 2015 Google Pixel */
 		.ident = "Chromebook Pixel 2",
 		.matches = {
-			DMI_MATCH(DMI_SYS_VENDOR, "GOOGLE"),
 			DMI_MATCH(DMI_PRODUCT_NAME, "Samus"),
 		},
 		.driver_data = samus_platform_data,
 	},
 	{
+		/* Acer C720P Chromebook */
+		.ident = "Acer C720",
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_NAME, "Peppy"),
+		},
+		.driver_data = peppy_platform_data,
+	},
+	{
 		/* Other Google Chromebooks */
 		.ident = "Chromebook",
 		.matches = {
@@ -2530,6 +2545,14 @@
 		},
 		.driver_data = chromebook_platform_data,
 	},
+	{
+		/* Other Google Chromebooks - upstream coreboot */
+		.ident = "Chromebook",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Google"),
+		},
+		.driver_data = chromebook_platform_data,
+	},
 	{ }
 };
 
